snp_motion_planning/src/planning_server.cpp

```diff
@@ -326,92 +346,206 @@ class PlanningServer
+    tesseract_planning::TaskComposerNode::UPtr task = factory.createTaskComposerNode(task_name);
+    if (!task)
+      throw std::runtime_error("Failed to create '" + task_name + "' task");
+
+    // Save dot graph
+    {
+      std::ofstream tc_out_data(tesseract_common::getTempPath() + task_name + ".dot");
+      task->dump(tc_out_data);
+    }
+
+    const std::string input_key = task->getInputKeys().front();
+    const std::string output_key = task->getOutputKeys().front();
+    auto task_data = std::make_shared<tesseract_planning::TaskComposerDataStorage>();
+    task_data->setData(input_key, program);
+    tesseract_planning::TaskComposerProblem::Ptr problem =
+        std::make_shared<tesseract_planning::PlanningTaskComposerProblem>(env_, profile_dict);
+    problem->dotgraph = true;
+
+    // Update log level for debugging
+    auto log_level = console_bridge::getLogLevel();
+    if (get<bool>(node_, VERBOSE_PARAM))
+    {
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_DEBUG);
+
+      // Dump dotgraphs of each task for reference
+      const YAML::Node& task_plugins = task_composer_config["task_composer_plugins"]["tasks"]["plugins"];
+      for (auto it = task_plugins.begin(); it != task_plugins.end(); ++it)
+      {
+        auto task_plugin_name = it->first.as<std::string>();
+        std::ofstream f(tesseract_common::getTempPath() + task_plugin_name + ".dot");
+        tesseract_planning::TaskComposerNode::Ptr task = factory.createTaskComposerNode(task_plugin_name);
+        if (!task)
+          throw std::runtime_error("Failed to load task: '" + task_plugin_name + "'");
+        task->dump(f);
+      }
+    }
+
+    // Run problem
+    tesseract_planning::TaskComposerFuture::UPtr result = executor->run(*task, problem, task_data);
+    result->wait();
+
+    // Save the output dot graph
+    {
+      std::ofstream tc_out_results(tesseract_common::getTempPath() + task_name + "_results.dot");
+      static_cast<const tesseract_planning::TaskComposerGraph&>(*task).dump(tc_out_results, nullptr,
+                                                                            result->context->task_infos.getInfoMap());
+    }
+
+    // Reset the log level
+    console_bridge::setLogLevel(log_level);
+
+    // Check for successful plan
+    if (!result->context->isSuccessful() || result->context->isAborted())
+      throw std::runtime_error("Failed to create motion plan");
+
+    // Get results of successful plan
+    tesseract_planning::CompositeInstruction program_results =
+        result->context->data_storage->getData(output_key).as<tesseract_planning::CompositeInstruction>();
+
+    // Send joint trajectory to Tesseract plotter widget
+    plotter_->plotTrajectory(toJointTrajectory(program_results), *env_->getStateSolver());
+
+    return program_results;
+  }
+
+  void planCallback(const snp_msgs::srv::GenerateMotionPlan::Request::SharedPtr req,
```

Comment:
Probably should change this name to be more descriptive now, like `planProcessCallback`

Reply:
Changed to `processMotionPlanCallback` to match the naming scheme for the freespace planner callback